src/Microsoft.AspNet.Security/AuthenticationTicket.cs

```diff
@@ -29,11 +29,33 @@ public AuthenticationTicket(ClaimsIdentity identity, AuthenticationProperties pr
         }
 
         /// <summary>
+        /// Initializes a new instance of the <see cref="AuthenticationTicket"/> class
+        /// </summary>
+        /// <param name="identity"></param>
+        /// <param name="properties"></param>
+        public AuthenticationTicket(ClaimsPrincipal principal, AuthenticationProperties properties, string authenticationType)
+        {
+            AuthenticationType = authenticationType;
+            Principal = principal;
+            Properties = properties ?? new AuthenticationProperties();
+        }
+
+        /// <summary>
+        /// Gets the authenticated user identity.
+        /// </summary>
+        public string AuthenticationType { get; private set; }
+
+        /// <summary>
         /// Gets the authenticated user identity.
         /// </summary>
         public ClaimsIdentity Identity { get; private set; }
```

Comment:
Hum, you mean the old `IClaimsPrincipal` used by WIF (`Microsoft.IdentityModel`)?
The `ClaimsPrincipal` from `mscorlib` doesn't implement this interface, that doesn't exist anymore in the .NET 4.5 port.